test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -10051,6 +10051,287 @@ BOOST_AUTO_TEST_CASE(delegatecall_return_value)
 	BOOST_CHECK(callContractFunction("get_delegated()") == encodeArgs(u256(1)));
 }
 
+BOOST_AUTO_TEST_CASE(snark)
+{
+	char const* sourceCode = R"(
+	library Pairing {
+		struct G1Point {
+			uint X;
+			uint Y;
+		}
+		// Encoding of field elements is: X[0] * z + X[1]
+		struct G2Point {
+			uint[2] X;
+			uint[2] Y;
+		}
+
+		/// @return the generator of G1
+		function P1() internal returns (G1Point) {
+			return G1Point(1, 2);
+		}
+
+		/// @return the generator of G2
+		function P2() internal returns (G2Point) {
+			return G2Point(
+				[11559732032986387107991004021392285783925812861821192530917403151452391805634,
+				 10857046999023057135944570762232829481370756359578518086990519993285655852781],
+				[4082367875863433681332203403145435568316851327593401208105741076214120093531,
+				 8495653923123431417604973247489272438418190587263600148770280649306958101930]
+			);
+		}
+
+		function negate(G1Point p) internal returns (G1Point) {
+			// The prime q in the base field F_q for G1
+			uint q = 21888242871839275222246405745257275088696311157297823662689037894645226208583;
+			if (p.X == 0 && p.Y == 0)
+				return G1Point(0, 0);
+			return G1Point(p.X, q - (p.Y % q));
+		}
+
+		function add(G1Point p1, G1Point p2) internal returns (G1Point r) {
+			uint[4] memory input;
+			input[0] = p1.X;
+			input[1] = p1.Y;
+			input[2] = p2.X;
+			input[3] = p2.Y;
+			bool success;
+			assembly {
+				success := call(sub(gas, 2000), 0x20, 0, input, 0xc0, r, 0x60)
```

Comment:
Could be `static_call`, if the precompiles are enabled the opcode should be available too.

Also shouldn't it check that `returndatasize` equals `0x60` too?